tests/mapper/test_graphwave_generator.py

```diff
@@ -0,0 +1,272 @@
+import networkx as nx
+import numpy as np
+import pytest
+import scipy.sparse as sps
+import tensorflow as tf
+
+
+class TestGraphWave:
+
+    gnx = nx.barbell_graph(m1=10, m2=11)
+
+    G = StellarGraph(gnx)
+    sample_points = np.linspace(0, 100, 50).astype(np.float32)
+    num_nodes = len(G.nodes())
+
+    def test_init(self):
+        generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=10)
+
+        assert np.isclose(generator.scales, (0.1, 2, 3, 4)).all()
+        assert generator.deg == 10
+        assert generator.coeffs.shape == (4, 10 + 1)
+        assert generator.laplacian.shape == (len(self.gnx.nodes), len(self.gnx.nodes))
+
+    def test_bad_init(self):
+
+        with pytest.raises(TypeError):
+            generator = GraphWaveGenerator(None, scales=(0.1, 2, 3, 4), deg=10)
+
+        with pytest.raises(TypeError, match="deg: expected.*found float"):
+            generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=1.1)
+
+        with pytest.raises(ValueError, match="deg: expected.*found 0"):
+            generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=0)
+
+    def test_bad_flow(self):
+        generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=10)
+
+        with pytest.raises(TypeError, match="batch_size: expected.*found float"):
+            generator.flow(self.G.nodes(), self.sample_points, batch_size=4.5)
+
+        with pytest.raises(ValueError, match="batch_size: expected.*found 0"):
+            generator.flow(self.G.nodes(), self.sample_points, batch_size=0)
+
+        with pytest.raises(TypeError, match="shuffle: expected.*found int"):
+            generator.flow(self.G.nodes(), self.sample_points, batch_size=1, shuffle=1)
+
+        with pytest.raises(TypeError, match="repeat: expected.*found int"):
+            generator.flow(self.G.nodes(), self.sample_points, batch_size=1, repeat=1)
+
+        with pytest.raises(
+            TypeError, match="num_parallel_calls: expected.*found float"
+        ):
+            generator.flow(
+                self.G.nodes(), self.sample_points, batch_size=1, num_parallel_calls=2.2
+            )
+
+        with pytest.raises(ValueError, match="num_parallel_calls: expected.*found 0"):
+            generator.flow(
+                self.G.nodes(), self.sample_points, batch_size=1, num_parallel_calls=0
+            )
+
+    @pytest.mark.parametrize("shuffle", [False, True])
+    def test_flow_shuffle(self, shuffle):
+
+        generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=10)
+
+        embeddings_dataset = generator.flow(
+            node_ids=self.G.nodes(),
+            sample_points=self.sample_points,
+            batch_size=1,
+            repeat=False,
+            shuffle=shuffle,
+        )
+
+        first, *rest = [
+            np.vstack([x.numpy() for x in embeddings_dataset]) for _ in range(20)
+        ]
+
+        if shuffle:
+            assert any((first == r).all() for r in rest) is False
```

Comment:
Also, here and all the other `(... == ...).all()` calls could be:

```suggestion
            assert not any(np.array_equal(first, r) for r in rest)
```

which is slightly more reliable, because it checks the shapes, and doesn't implicitly broadcast. (This would catch problems where the shapes of the LHS and RHS were very different, which is unlikely in this particular case, but it seems good to default to catching it.)